File: gitdot-web/packages/api/src/endpoint/repository/get-repository-resources.ts
import { z } from "zod";
import { RepositoryResourcesResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetRepositoryResourcesRequest = z.object({});
export type GetRepositoryResourcesRequest = z.infer<
  typeof GetRepositoryResourcesRequest
>;

export const GetRepositoryResourcesResponse = RepositoryResourcesResource;
export type GetRepositoryResourcesResponse = z.infer<
  typeof GetRepositoryResourcesResponse
>;

export const GetRepositoryResources = {
  path: "/repository/{owner}/{repo}/resources",
  method: "POST",
  request: GetRepositoryResourcesRequest,
  response: GetRepositoryResourcesResponse,
} as const satisfies Endpoint;
export type GetRepositoryResources = typeof GetRepositoryResources;

File: gitdot-api/src/endpoint/repository/get_repository_resources.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryResourcesResource};

pub struct GetRepositoryResources;

impl Endpoint for GetRepositoryResources {
    const PATH: &'static str = "/repository/{owner}/{repo}/resources";
    const METHOD: http::Method = http::Method::POST;

    type Request = GetRepositoryResourcesRequest;
    type Response = GetRepositoryResourcesResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetRepositoryResourcesRequest {}

pub type GetRepositoryResourcesResponse = RepositoryResourcesResource;
